api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -126,16 +133,19 @@ public OrderGroup saveOrderGroup(OrderGroup orderGroup) throws APIException {
 	public synchronized Order saveRetrospectiveOrder(Order order, OrderContext orderContext) {
 		return saveOrder(order, orderContext, true);
 	}
-	
-	private Order saveOrder(Order order, OrderContext orderContext, boolean isRetrospective) {
-		if (order.getOrderId() != null) {
-			throw new UnchangeableObjectException("Order.cannot.edit.existing");
-		}
+
+	private void failOnExistingOrder(Order order) {
```

Comment:
@Nuwantha I just checked the issue on jira, sorry for this but since the issue was created the line numbers must have changed.

the `failOnExistingOrder(Order order)` is of course not for setting the dateActivated but for the first if block that throws the UnchangeableObjectException